python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -125,13 +125,35 @@ def render_movie_in_nuke(self, path, output_path,
                              first_frame, last_frame,
                              version, name,
                              color_space,
-                             replace_data,
+                             fields=None,
                              active_progress_info=None):
+        """
+        Renders the movie using a Nuke subprocess,
+        along with slate/burnins using all the app settings.
+
+        :param path:            The path where frames should be found.
+        :param output_path:     The path where the movie should be written to
+        :param width:           Movie width
+        :param height:          Movie height
+        :param first_frame:     The first frame of the sequence of frames
+        :param last_frame:      The last frame of the sequence of frames
+        :param version:         Version currently being published
+        :param name:            Name of the file being published
+        :param color_space:     Colorspace used to create the frames
+        :param fields:          Any additional information to be used in slate/burnins
+        :param active_progress_info: Any function that receives the progress percentage
+                                     Can be used to update GUI
+        """
+        # add to information passed for preprocessing
+        fields["first_frame"] = first_frame
+        fields["last_frame"] = last_frame
+        fields["path"] = path
```

Comment:
Moved as much replace data creation as possible to hook, but packaging what information is already available here to be passed to the hook.